### PR TITLE
Dịch "softmax-regression-gluon_vn" - Phần 3

### LINKED ISSUE #766: Dịch "softmax-regression-gluon_vn" - Phần 3
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression-gluon_vn.md#L126-L187

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tamnguyenvan**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #766

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 1]
```diff
@@ -133,14 +133,15 @@ loss = gluon.loss.SoftmaxCrossEntropyLoss()
...
+## Thuật toán tối ưu
 
 <!--
 Here, we use minibatch stochastic gradient descent with a learning rate of $0.1$ as the optimization algorithm.
 Note that this is the same as we applied in the linear regression example and it illustrates the general applicability of the optimizers.
 -->
 
-*dịch đoạn phía trên*
+Ở đây, chúng ta sử dụng thuật toán tối ưu hạ gradient ngẫu nhiên theo minibatch với tốc độ học bằng $0.1$.
+Lưu ý rằng việc này giống với khi áp dụng cho bài toán hồi quy tuyến tính, nó chứng minh khả năng ứng dụng tổng quát của phương pháp tối ưu này.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Lưu ý rằng đây cũng là thuật toán áp dụng cho bài toán hồi quy tuyến tính, và điều này chứng minh khả năng ứng dụng tổng quát của các phương pháp tối ưu.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 2]
```diff
@@ -150,13 +151,13 @@ trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.1})
...
 
-## *dịch tiêu đề phía trên*
+## Huấn luyện
 
 <!--
 Next we call the training function defined in the last section to train a model.
 -->
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta sẽ gọi tới hàm huấn luyện đã được khai báo ở mục trước để huấn luyện mô hình.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Tiếp theo, chúng ta sẽ gọi hàm huấn luyện đã được khai báo ở mục trước để huấn luyện mô hình.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 3]
```diff
@@ -169,20 +170,22 @@ Note that in many cases, Gluon takes additional precautions beyond these most we
 saving us from even more pitfalls that we would encounter if we tried to code all of our models from scratch in practice.
 -->
 
-*dịch đoạn phía trên*
+Giống như trước đây, thuật toán này cũng hội tụ tới một trạng thái có độ chính xác 83.7%, trừ một điều là với ít dòng mã hơn.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Giống như trước, thuật toán này cũng hội tụ tới nghiệm cho độ chính xác 83.7%, ngoại trừ bây giờ ta viết ít dòng mã hơn.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 4]
```diff
@@ -169,20 +170,22 @@ Note that in many cases, Gluon takes additional precautions beyond these most we
...
+Lưu ý rằng trong nhiều trường hợp, Gluon thực hiện thêm một số phòng tránh ngoài các thủ thuật đã biết để đảm bảo không xảy ra tràn số, giúp ta tránh gặp phải các lỗi khi cố gắng lập trình toàn bộ mô hình từ đầu.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Lưu ý rằng trong nhiều trường hợp, Gluon thực hiện thêm một số biện pháp phòng tránh khác với các thủ thuật phổ biến để đảm bảo không xảy ra tràn số, giúp ta tránh mắc lỗi khi đang cố gắng lập trình toàn bộ các mô hình từ đầu.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 5]
```diff
@@ -169,20 +170,22 @@ Note that in many cases, Gluon takes additional precautions beyond these most we
...
+## Bài tập
 
 <!--
 1. Try adjusting the hyper-parameters, such as batch size, epoch, and learning rate, to see what the results are.
 2. Why might the test accuracy decrease again after a while? How could we fix this?
 -->
 
-*dịch đoạn phía trên*
+1. Thử thay đổi các siêu tham số, như là kích thước batch, epoch, và tốc độ học, để xem kết quả như thế nào.
+2. Tại sao độ chính xác khi kiểm tra lại bị giảm sau một khoảng thời gian? Chúng ta có thể sửa lại bằng cách nào.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
2. Tại sao độ chính xác trên tập kiểm tra lại giảm sau một khoảng thời gian? Chúng ta giải quyết việc này thế nào?
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 1 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Lưu ý rằng cách làm này y chang cách làm ở ví dụ về hồi quy tuyến tính, minh chứng cho tính tổng quát của phương pháp tối ưu hạ gradient ngẫu nhiên.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 3 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Giống lần trước, thuật toán hội tụ và nghiệm có độ chính xác 83.7%, chỉ khác một điều là cần ít dòng code hơn.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 4 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Lưu ý rằng trong nhiều trường hợp, Gluon giảm thiểu các sai số tính toán bằng cách sử dụng các kỹ thuật phổ biến và ít phổ biến giúp tránh khỏi nhiều lỗi kỹ thuật so với khi ta tự lập trình xây dựng mô hình từ đầu.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 1 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: @tanduong nguyên văn là `optimizers`, ý tác giả nói các thuật toán tối ưu đều áp dụng được cho các mô hình khác nhau

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 3 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: mình thấy của suggestion của @tanduong ổn hơn

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 4 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: meh mình thấy vẫn chưa mượt lắm, chờ ý kiến của ng khác vậy

**Comment by tamnguyenvan** [comment]
Cảm ơn @tanduong @minhduc0711 đã review. Hôm qua dịch trong lúc hơi ngáo ngơ :3

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 1 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: `optimizers` ở đây chỉ các `bộ tối ưu` trong các bài toán khác nhau, nhưng đều dùng SGD ấy. Mình thấy gợi ý của @tanduong đúng hơn.

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 3 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: Mình sửa suggestion của @tanduong, @tamnguyenvan có thể commit trực tiếp luôn nhé.

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 4 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]:
```suggestion
Lưu ý rằng trong nhiều trường hợp, Gluon tự động sử dụng các kỹ thuật phổ biến để tránh các lỗi kĩ thuật tính toán mà ta dễ gặp phải nếu tự xây dựng mô hình từ đầu.

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: @lkhphuc thiếu từ `beyond` mất rồi
```suggestion
Lưu ý rằng trong nhiều trường hợp, Gluon không chỉ dùng các mánh phổ biến mà còn sử dụng các kỹ thuật khác để tránh các lỗi kĩ thuật tính toán mà ta dễ gặp phải nếu tự xây dựng mô hình từ đầu.
```

File: chapter_linear-networks/softmax-regression-gluon_vn.md (hunk 1 above)
Comment (tiepvupsu) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: Đúng rồi `optimizer` là `bộ tối ưu` chứ không phải là `phương pháp tối ưu`.

Comment (tiepvupsu) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: @tanduong từ `y chang` nghe không được "academic writing" cho lắm, `giống hệt` quen thuộc hơn.

Comment (tanduong) [reply chapter_linear-networks/softmax-regression-gluon_vn.md]: Mình đã sửa suggestion theo 2 nhận xét trên.

**Comment by tamnguyenvan** [comment]
Cảm ơn tất cả mọi người đã review :D

File: chapter_linear-networks/softmax-regression-gluon_vn.md [hunk 6]
```diff
@@ -169,20 +170,23 @@ Note that in many cases, Gluon takes additional precautions beyond these most we
...
 
 <!--
 1. Try adjusting the hyper-parameters, such as batch size, epoch, and learning rate, to see what the results are.
 2. Why might the test accuracy decrease again after a while? How could we fix this?
 -->
 
-*dịch đoạn phía trên*
+1. Thử thay đổi các siêu tham số, như là kích thước batch, epoch, và tốc độ học, để xem kết quả như thế nào.
+2. Tại sao độ chính xác trên tập kiểm tra lại giảm sau một khoảng thời gian? 
+Chúng ta giải quyết việc này thế nào?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-gluon_vn.md]: hôm trước mình sửa nhầm :))
```suggestion
2. Tại sao độ chính xác trên tập kiểm tra lại giảm sau một khoảng thời gian? Chúng ta giải quyết việc này thế nào?
```
